mysql-test/suite/innodb/t/instant_alter_charset.test

```diff
@@ -0,0 +1,411 @@
+--source include/have_innodb.inc
+--source include/innodb_row_format.inc
```

Comment:
First of all, `innodb_row_format.inc` * `innodb_page_size.inc` * `alter_algorithm.inc` = 60 which is not fun. Also I don't know why it's needed. Could you elaborate on that? `COPY` is used where it's needed for reference f.ex in `latin1_swedish_special_case` tests.

Reply:
OK, maybe `innodb_page_size.inc` is not that useful for this change. `mtr` is primarily a regression test suite, not a stress test. Page sizes should be covered by additional stress testing. They are interesting, because they affect page splits.

I still think we should cover both COPY and INPLACE with `.combinations` and `.rdiff` files, for all 3 `innodb_row_format.combinations`. That would be only 6 combinations.

Because NOCOPY and INSTANT are subsets of INPLACE, I do not see a need to test with them. We would prove with `innodb_instant_alter_column` that every operation was instantaneous unless the COPY algorithm was used.